luskan/src/test/java/org/opencadc/luskan/SiteLocationsConverterTest.java

```diff
@@ -0,0 +1,171 @@
+ *  modify it under the terms of         modifier suivant les termes de
+ *  the GNU Affero General Public        la “GNU Affero General Public
+ *  License as published by the          License” telle que publiée
+ *  Free Software Foundation,            par la Free Software Foundation
+ *  either version 3 of the              : soit la version 3 de cette
+ *  License, or (at your option)         licence, soit (à votre gré)
+ *  any later version.                   toute version ultérieure.
+ *
+ *  OpenCADC is distributed in the       OpenCADC est distribué
+ *  hope that it will be useful,         dans l’espoir qu’il vous
+ *  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+ *  without even the implied             GARANTIE : sans même la garantie
+ *  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+ *  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+ *  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+ *  General Public License for           Générale Publique GNU Affero
+ *  more details.                        pour plus de détails.
+ *
+ *  You should have received             Vous devriez avoir reçu une
+ *  a copy of the GNU Affero             copie de la Licence Générale
+ *  General Public License along         Publique GNU Affero avec
+ *  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+ *  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+ *                                       <http://www.gnu.org/licenses/>.
+ *
+ *  : 5 $
+ *
+ ************************************************************************
+ */
+
+package org.opencadc.luskan;
+
+import ca.nrc.cadc.tap.TapQuery;
+import ca.nrc.cadc.tap.schema.TapSchema;
+import ca.nrc.cadc.util.Log4jInit;
+import ca.nrc.cadc.util.MultiValuedProperties;
+import ca.nrc.cadc.uws.Parameter;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class SiteLocationsConverterTest {
+    private static final Logger log = Logger.getLogger(SiteLocationsConverterTest.class);
+
+    static {
+        Log4jInit.setLevel("org.opencadc.luskan", Level.INFO);
+    }
+
+    private static final TapSchema tapSchema = TestUtil.loadTapSchema();
+
+    private static final String selectList =
+        "inventory.artifact.uri, inventory.artifact.uribucket, inventory.artifact.contentchecksum, "
+            + "inventory.artifact.contentLastmodified, inventory.artifact.contentlength, "
+            + "inventory.artifact.contenttype, inventory.artifact.contentencoding, inventory.artifact.lastmodified, "
+            + "inventory.artifact.metachecksum, inventory.artifact.id ";
+    private static final String fromClause = "FROM inventory.artifact";
+    private static final String isNotNullConstraint = "(inventory.artifact.sitelocations IS NOT NULL)";
+
```

Comment:
The relevant variants to test here are:
1. vanilla: FROM inventory.Artifact --> ({column} IS NOT NULL}
2. table alias: FROM  inventory.Artifact AS a --> (a.{column} IS NOT NULL)

So you have to find the table reference in the from clause and figure out whether or not to use the alias.

3. join: FROM inventory.Artifact a join inventory.Artifact b on a.id=b.id

Ok, it's hard to find a useful thing to do with that, but it should add a constraint for each alias)

Reply:
Updated unit tests with the above cases.